### ISSUE TITLE
Wrong numbers for home consumption.

### ISSUE THREAD

**Issue by Cyrelion** [flixlix/power-flow-card-plus#231]
Wrong numbers for home consumption.

Diagram shows 3kW but consumption is only 700 W

![IMG_1881](https://github.com/flixlix/power-flow-card-plus/assets/16318570/173e6fdf-75d0-4828-b4dc-6c6a76f4d82b)
![IMG_1882](https://github.com/flixlix/power-flow-card-plus/assets/16318570/ee1acb58-cdcb-444b-916b-d5929b890978)

**Comment by Frank-H-H** [comment]
I am not quite sure, if I have the same issue as above, but it looks like that. From my understanding, "Home" is supposed to show the actual consumption within the house in total. If you also have a loading, your home still uses up all that incoming solar energy, but that number would already be visualized by the battery indicator. The "Home" in my perspective should therefore not include the battery loading consumption again. Have a look at the following image: I've got 45W from grid and 940W from Solar. My battery is loading with 695W. My "Home" consumption is computed to be 985W, while in fact it is more of 290W (Grid + Solar - BatteryConsumption).

![grafik](https://github.com/flixlix/power-flow-card-plus/assets/76811133/499b4f2e-ecbb-4f95-b00c-7930c140de72)

**Comment by Cyrelion** [comment]
Yes looks like home now includes the battery loading. That has changed from 1.3 to 1.4

**Comment by Frank-H-H** [comment]
This looks like the place requiring the update: 
https://github.com/flixlix/power-flow-card-plus/blob/6b551dd532bc32aa06eadba870f6e4aefb7f1499/src/power-flow-card-plus.ts#L706
It should probably also have an "- toBattery"?

**Comment by JayMcG** [comment]
Hi. Also seeing the same issue. The battery charging value from grid seems to be going into Home as well as battery. My total Home load is 2596w.

![Capture](https://github.com/flixlix/power-flow-card-plus/assets/17012684/3357e12d-a3ea-43bb-9e54-d908d46d0477)

**Comment by turbolooser** [comment]
Same issue here. I think this new way of calculating the home consumption should be optional/configurable.

**Comment by dolby85** [comment]
same problem here, with previous version 1.3 issue wasn't present.

**Comment by elektrik216** [comment]
I also have reverted back to 1.3. My opinion is that charging battery should be separate, not included in home consumption, because latter it is gonna be consumed anyway in home. In this way is some kind of double consumption the same energy.

**Comment by juscode** [comment]
Same issue here would really like this changed back, is there an easy way to revert to 1.3? I'm quite new to HA.

**Comment by MattJeanes** [comment]
If you have your home entity set that has the correct value you can set `override_state: true` on the `home` section and it will use that as a workaround.

**Comment by juscode** [comment]
Thanks

**Comment by malrogers7** [comment]
Same issue here on 1.4, the Home entity has started including the power used to charge the battery.
Have used @MattJeanes suggestion of override_state: true. Thanks 👍

**Comment by flixlix** [comment]
As mentioned in #227, please try this file and see if it fixes the issue. I think it should also fix this issue

# Possibility of Fix
Created #241 
Can you check if using this file manually fixes the issue?
FIY: This should be returned in the console. If not you do not have this version
![console.log](https://github.com/flixlix/power-flow-card-plus/assets/61006057/31de53ba-0ba8-4b5f-b24f-7f484d33832c)
[power-flow-card-plus.js.zip](https://github.com/flixlix/power-flow-card-plus/files/11619227/power-flow-card-plus.js.zip)

**Comment by Frank-H-H** [comment]
It took me some time to figure out how to test the provided file (manually overwriting the file /config/www/community/power-flow-card-plus/power-flow-card-plus.js still would cause the old file to load for whatever reason. I placed the provided js file with a different name into that folder and linked the resource manually in the dashboard).

However, it looks like this fixes the issue for me. Weather situation is similar to yesterday. I know, that my current home consumption should be in the range of 100 - to 250 watts and now this is reflected in the card properly without taking the battery charge into account. Proof for using the correct version is also attached:

![grafik](https://github.com/flixlix/power-flow-card-plus/assets/76811133/de0cc731-cbac-4e17-82b7-fe53d9103980)

**Comment by kumarunster** [comment]
same here
<img width="677" alt="Bildschirmfoto 2023-06-01 um 09 29 44" src="https://github.com/flixlix/power-flow-card-plus/assets/2104820/04bfe405-8018-4b53-b471-16007d82d6d2">
<img width="543" alt="Bildschirmfoto 2023-06-01 um 09 30 22" src="https://github.com/flixlix/power-flow-card-plus/assets/2104820/fb70e3f2-e6eb-436a-905b-f2e43462b9cf">
<img width="453" alt="Bildschirmfoto 2023-06-01 um 09 32 43" src="https://github.com/flixlix/power-flow-card-plus/assets/2104820/406e500e-ab9e-4a3b-a52b-cabfa7eedc95">

it should the value provided in the config in my opinion and not try to calculate some numbers itself

**Comment by malrogers7** [comment]
@kumarunster 
> it should show the value provided in the config, in my opinion, and not try to calculate some numbers itself

Use `override_state: true` on the `home` section and it will use the entity set in the `home` section

My code:
```
  home:
    entity: sensor.load_power_calculated
    override_state: true
```

**Comment by Frank-H-H** [comment]
@juscode : 
- Save the js file in the provided zip in the same folder as where the default file is, but use a different name (I used /config/www/community/power-flow-card-plus/p.js)
- Open your dashboard in edit mode. Then click the three dots on the top right and then edit resources or so
![grafik](https://github.com/flixlix/power-flow-card-plus/assets/76811133/babea534-8d80-41e3-b3eb-68d9903718b2)

- You should see a list of loaded js files. Select the one from power flow and (first back up the current value and then) edit the file name whatever you saved the file as (I used /hacsfiles/power-flow-card-plus/p.js?hacstag=...)
- After closing those dialogs you need to do a force refresh of the browser (CTRL + R should do that)

**Comment by flixlix** [comment]
Sorry to everyone for the inconvenience. This issue should now be fixed and will be available in the next version.

**Comment by Frank-H-H** [comment]
Same here: Thanks for all the work you are putting into that :)

**Comment by MattJeanes** [comment]
Can confirm that it's working correctly on the latest version. Thank you!

**Comment by juscode** [comment]
Thanks for the new version ! just one quick question is there a way to make the text in the grid / batteries appear black instead of the dynamic colour ?

**Comment by flixlix** [comment]
@juscode that should be the default behavior when `color_value` is not defined

**Comment by juscode** [comment]
The default colour is always match the circle colour, ideally I wanted to set it to black regardless of the circle colour so its easier to read the energy. Is this a bug then?

**Comment by flixlix** [comment]
Ohhh, now I get what you mean!
Yeah no, this is not currently a feature.
It would be a Feature Request.

But these colors are there for a reason: To more quickly connect the lines to the states semantically, so I don't know how accessible it would be to have an all black text in there, since you wouldn't know from a glance which one is production and which one is consumption

**Comment by juscode** [comment]
Yes I do see what you mean, will stick with it as is for now then ! thanks again for sorting it all !

**Comment by benvanmierloo** [comment]
Hi, 

it seems I have found another edge case linked to this issue. 

When feeding all power (Solar + Battery) to the grid, the Home calculation not correct. In the screenshot below you can see the situation. 4.67 kW is being fed into the grid. The battery supplies 2.62 kW and the Solar panels 2.38 kW. This means the Home consumption should be Solar + Battery discharge - Grid injection (2.38kW+2.62kW-4.67kW) = 330W instead of 2.62kW,

![screenshot_discharge_to_grid](https://github.com/flixlix/power-flow-card-plus/assets/3785514/81ccef70-5200-4444-85c9-ccd21ae9db2f)


regards
Ben
